src/main/java/io/seqera/tower/cli/commands/data/links/AbstractDataLinksCmd.java

```diff
@@ -0,0 +1,111 @@
+import io.seqera.tower.model.DataLinkType;
+import io.seqera.tower.model.DataLinkUpdateRequest;
+
+import java.util.Objects;
+
+public class AbstractDataLinksCmd extends AbstractApiCmd {
+
+    public static String buildSearch(String name, String providers, String region, String uri) {
+        StringBuilder builder = new StringBuilder();
+        if (name != null && !name.isBlank()) {
+            builder.append(name);
+        }
+        if (providers != null && !providers.isBlank()) {
+            appendParameter(builder, providers, "provider");
+        }
+        if (region != null && !region.isBlank()) {
+            appendParameter(builder, region, "region");
+        }
+        if (uri != null && !uri.isBlank()) {
+            appendParameter(builder, uri, "resourceRef");
+        }
+        if (builder.length() > 0)
+            return builder.toString();
+        return null;
+    }
+
+    private static void appendParameter(StringBuilder input, String param, String paramName) {
+        if (param != null && !param.isBlank()) {
+            if (input.length() > 0) {
+                input.append(" ");
+            }
+            input.append(paramName).append(":").append(param);
+        }
+    }
+
+    DataLinksFetchStatus checkDataLinksFetchStatus(Long wspId, String credentialsId) {
+        int status = 0;
+        try {
+            status = api().listDataLinksWithHttpInfo(wspId, credentialsId, null, 1, 0, null).getStatusCode();
+        } catch (ApiException e) {
+            return DataLinksFetchStatus.ERROR;
+        }
+        switch (status) {
+            case 200:
+                return DataLinksFetchStatus.DONE;
+            case 202:
+                return DataLinksFetchStatus.FETCHING;
+            default:
+                return DataLinksFetchStatus.ERROR;
+        }
+    }
+
+    DataLinkDto addDataLink(Long wspId, String name, String description, String url, String provider, String credsId) throws ApiException {
+        DataLinkCreateRequest req = new DataLinkCreateRequest();
+        req.name(name);
+        req.description(description);
+        req.resourceRef(url);
+        req.type(DataLinkType.BUCKET);
+        req.provider(DataLinkProvider.fromValue(provider.toLowerCase()));
+
+        if (Objects.isNull(credsId)) {
+            req.publicAccessible(true);
+        } else {
+            req.publicAccessible(false);
+            req.credentialsId(credsId);
+        }
+
+        return api().createCustomDataLink(req, wspId);
+    }
+
+    DataLinkDto updateDataLink(Long wspId, String id, String name, String description, String credsId) throws ApiException {
+        DataLinkUpdateRequest req = new DataLinkUpdateRequest();
+        if (name != null)
+            req.name(name);
+        if (description != null)
+            req.description(description);
+        if (credsId != null)
+            req.credentialsId(credsId);
+
+        return api().updateCustomDataLink(id, req, wspId);
+    }
```

Comment:
Is there a strong reason for having these methods in the `AbstractDataLinksCmd` class instead of their respective commands (`AddCmd`, `UpdateCmd`)?